src/py/flwr/server/state/sqlite_state.py

```diff
@@ -457,13 +471,35 @@ def unregister_node(self, node_id: int) -> None:
         query = "DELETE FROM node WHERE node_id = :node_id;"
         self.query(query, {"node_id": node_id})
 
-    def get_nodes(self) -> Set[int]:
+    def get_nodes(self, workload_id: str) -> Set[int]:
         """Retrieve all currently stored node IDs as a set."""
+        # Validate workload ID
+        query = "SELECT COUNT(*) FROM workload WHERE workload_id = ?;"
+        if self.query(query, (workload_id,))[0]["COUNT(*)"] == 0:
+            log(ERROR, "`workload_id` is invalid")
+            return set()
+
+        # Get nodes
         query = "SELECT * FROM node;"
         rows = self.query(query)
         result: Set[int] = {row["node_id"] for row in rows}
         return result
 
+    def create_workload(self) -> str:
```

Comment:
We don't let the state create the `node_id` and handle that responsibility outside of the state module. Should we do the same here? I feel like it should be unified.

Reply:
I am thinking of the same issue. We can also pick a random integer as workload_id in the DriverAPI and include that in the `CreateWorkloadRequest`. @danieljanes which option would you prefer, sample `workload_id` on the driver or on the server?

Reply:
@panh99 @tanertopal let's use the same approach that we use for `node_id`